src/nsrt_learning/side_predicate_learning.py

```diff
@@ -455,70 +382,193 @@ def _sideline(self) -> List[PartialNSRTAndDatastore]:
                 option = segment.get_option()
                 # Find the PNAD associated with this option.
                 pnad = param_opt_to_pnad[option.parent]
-                var_to_obj = pnad.get_sub_for_member_segment(segment)
                 # Compute the ground atoms that must be added on this timestep.
+                # Note that they must be a subset of the add effects.
                 necessary_add_effects = necessary_image - atoms_seq[t]
-                # Compute the add effects that this PNAD currently predicts.
-                predicted_add_effects = {
-                    a.ground(var_to_obj)
-                    for a in pnad.op.add_effects
-                }
-                # If the predicted add effects are missing necessary add
-                # effects, then we need to make the add effects more specific.
-                if not necessary_add_effects.issubset(predicted_add_effects):
-                    obj_to_var = {o: v for v, o in var_to_obj.items()}
-                    assert len(obj_to_var) == len(var_to_obj)
+                assert necessary_add_effects.issubset(segment.add_effects)
+
+                # Find a mapping from the variables in the PNAD add effects
+                # and option to the objects in necessary_add_effects and the
+                # segment's option. If one exists, we don't need to modify this
+                # PNAD. Otherwise, we must make its add effects more specific.
```

Comment:
Could we add a note to this comment that we assume all variables in the parameters of the PNAD must appear in the option args or add effects?